src/main/lombok/org/heigit/ohsome/ohsomeapi/executor/ExecutionUtils.java

```diff
@@ -600,10 +597,10 @@ public void writeCsvResponse(Result[] resultSet, HttpServletResponse servletResp
    */
   public static Pair<Pair<Integer, Integer>, OSMEntitySnapshot> mapSnapshotToTags(int keysInt,
       Integer[] valuesInt, OSMEntitySnapshot f) {
-    int[] tags = f.getEntity().getRawTags();
-    for (int i = 0; i < tags.length; i += 2) {
-      int tagKeyId = tags[i];
-      int tagValueId = tags[i + 1];
+    Iterable<OSHDBTag> tags = f.getEntity().getTags();
+    for (OSHDBTag tag : tags) {
+      int tagKeyId = tag.getKey();
+      int tagValueId = tag.getValue();
       if (tagKeyId == keysInt) {
         if (valuesInt.length == 0) {
           return new ImmutablePair<>(new ImmutablePair<>(tagKeyId, tagValueId), f);
```

Comment:
Why not just take the `OSHDBTag` here instead of creating a new `ImmutablePair(tagKeyId,  tagValueId)`.

Reply:
makes sense, will do :+1:

Reply:
well, actually, the values returned by this method are later required to be Comparable, which `OSHDBTag` currently isn't. Since fixing that wouldn't be a trivial change, I'd rather leave the current implementation as it is, and deal with it later.